### ISSUE TITLE
cuda zero-copy device to device (intra host or inter host)

### ISSUE THREAD

**Issue by abellina** [openucx/ucx#4432]
Is there currently a way to get a buffer allocated via cuda copied to either:

1. The same machine, different GPU
2. Different machine

Without a copy to the host memory?

**Comment by bureddy** [comment]
Yes if you have a transport Like IB that can do zero-copy transfer from CUDA memory using GPUDirectRDMA.
you need to set  following parameters
```
 -x UCX_RNDV_SCHEME=put_zcopy  
-x  UCX_RNDV_THRESH=1
```

is your GPUs with in the same machine are  Cuda IPC accessible?

**Comment by abellina** [comment]
Yes to both questions. I will test IB also, does Cuda IPC just get enabled via `UCX_TLS=cuda_ipc`? Do I need IB for it?

Currently I am running a test where I send and received tagged between two processes in the same machine, with send/recv buffers in two different gpus connected over PCIe.
